TopMovies/MoviesConnector/TopMoviesConnector.swift

```diff
@@ -16,21 +18,31 @@ class TopMoviesConnector: BaseConnector<TopMoviesProps> {
   override func newState(state: MainState) {
     switch state.movieCategoriesState.categoriesList {
     case let .completed(categoriesID):
-      let props =
-        TopMoviesProps(
-          movieCategories:
-            categoriesID
-            .compactMap{ MovieCategoryProps(
-              categoryNameText:
-                state.movieCategoriesState.relational[$0]?.title,
-              movies:
-                state.categoriesPaginationState.paginated[$0]?.list
-                .compactMap{ MovieCollectionProps(
-                  movie: state.moviesState.relational[$0]
-                ) } ?? [] )
-            }
-        )
-      _updateProps(props)
+      let updatedCategories = categoriesID
+        .reduce(into: [MovieCategory.ID: [Movie.ID]]()) { dict, categoryID in
+          var list = state.categoriesPaginationState.paginated[categoryID]?.list ?? []
+          list.removeLast(list.count - numberOfVisiableMoviesInCategory)
+          dict[categoryID] = list
+        }
+      if updatedCategories != alreadyShown {
```

Comment:
Comparing dictionaries is a very heavy operation, so I'd consider choosing another way for that.

Reply:
I'd prefer to store and compare ids, since you may have 5 ids just like before, but their order could've been changed.